pyinteraph/centrality_analysis.py

```diff
@@ -16,26 +16,96 @@ def get_hubs(G, **kwargs):
     hubs = {n : (d if d >= kwargs["hub"] else 0) for n, d in degree_tuple}
     return hubs
 
+def get_connected_components(G, **kwargs):
+    """Returns a dictionary of connected component index for each node.
+    The index is sorted in descending order of size of the component.
+    """
+    components = nx.algorithms.components.connected_components(G)
+    # reverse sort components by size
+    components = sorted(components, key = lambda x: len(x), reverse = True)
+    # get dict of component indexes
+    component_dict = {node : i + 1 for i, component in enumerate(components) \
+                                        for node in component}
+    # sort dictionary by node
+    sorted_dict = {node : component_dict[node] for node in G.nodes()}
+    return sorted_dict
+
 def get_degree_cent(G, **kwargs):
     """Returns a dictionary of degree centrality values for each node."""
 
     centrality_dict = nxc.degree_centrality(G)
     return centrality_dict
 
+def get_graph_without_glycine(G, identifiers, residue_names):
+    """Takes in a graph, a list of nodes for the graph and a list of
+    residues corresponding to each node. Returns a graph without any 
+    nodes corresponding to Glycine.
+    """
+    # get dictionary of node:res
+    node_dict = dict(zip(identifiers, residue_names))
+    # get all nodes which correspond to glycine
+    glycine_nodes = [node for node, res in node_dict.items() if res == "GLY"]
+    # create new graph with glycines
```

Comment:
Correct the comment: "create new graph without glycines", not "create new graph with glycines".